plutus-tx/src/PlutusTx/Blueprint/Contract.hs

```diff
@@ -99,7 +97,14 @@ instance ToJSON (ContractBlueprint referencedTypes) where
         . requiredField "preamble" contractPreamble
         . requiredField "validators" contractValidators
         . optionalField "$id" contractId
-        . optionalField "definitions" (ensure (not . Map.null) contractDefinitions)
+        . optionalField "definitions" definitions
    where
     schemaUrl :: String
     schemaUrl = "https://cips.cardano.org/cips/cip57/schemas/plutus-blueprint.json"
+
+    definitions :: Maybe (Map DefinitionId Aeson.Value)
+    definitions = ensure (not . Map.null) (definitionsToMap contractDefinitions toJSON)
+
+-- | 'Blueprint' wrapper to hide the 'referencedTypes' parameter as an existential type.
```

Comment:
why does this need to be a wrapper? Why not just change the definition of `ContractBlueprint`?